src/main/java/teammates/storage/api/CommentsDb.java

```diff
@@ -560,15 +540,16 @@ public CommentSearchResultBundle search(String queryString, List<InstructorAttri
         return getCommentsWithoutDeletedEntity(commentList);
     }
 
-    private List<Comment> getCommentEntitiesForGiver(String courseId, String giverEmail) {
+    private QueryWithParams getCommentsForGiverQuery(String courseId, String giverEmail) {
         Query q = getPm().newQuery(Comment.class);
         q.declareParameters("String courseIdParam, String giverEmailParam");
         q.setFilter("courseId == courseIdParam && giverEmail == giverEmailParam");
+        return new QueryWithParams(q, new Object[] {courseId, giverEmail});
+    }
 
-        @SuppressWarnings("unchecked")
-        List<Comment> commentList = (List<Comment>) q.execute(courseId, giverEmail);
-
-        return getCommentsWithoutDeletedEntity(commentList);
+    @SuppressWarnings("unchecked")
+    private List<Comment> getCommentEntitiesForGiver(String courseId, String giverEmail) {
+        return (List<Comment>) getCommentsForGiverQuery(courseId, giverEmail).execute();
```

Comment:
From what I understand from reading the docs, `JDOHelper.isDeleted` is only applicable for transactions, which we don't use at all.